File: docs/examples/jdbc-agent-template/src/test/java/com/dbx/agent/template/TemplateAgentTest.java
package com.dbx.agent.template;

import com.dbx.agent.test.TestSupport;
import com.dbx.agent.ExecuteQueryOptions;
import com.dbx.agent.test.JdbcAgentFake;
import java.util.List;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class TemplateAgentTest {
    @Test
    void executeQueryUsesStatementExecute() {
        TemplateAgent agent = new TemplateAgent();
        TestSupport.setPrivateConnection(agent, JdbcAgentFake.connection());

        agent.executeQuery("SHOW TABLES", null, new ExecuteQueryOptions());

        Assertions.assertEquals(
            List.of("setMaxRows:10001", "execute"),
            JdbcAgentFake.calls
        );
    }
}

File: docs/examples/jdbc-agent-template/src/test/kotlin/com/dbx/agent/template/TemplateAgentTest.kt
package com.dbx.agent.template

import com.dbx.agent.test.JdbcAgentFake
import com.dbx.agent.test.setPrivateConnection
import kotlin.test.Test
import kotlin.test.assertEquals

class TemplateAgentTest {
    @Test
    fun `executeQuery uses Statement execute`() {
        val agent = TemplateAgent()
        setPrivateConnection(agent, JdbcAgentFake.connection())

        agent.executeQuery("SHOW TABLES", null)

        assertEquals(listOf("execute"), JdbcAgentFake.calls)
    }
}
